components/datetime/src/mock/mod.rs

```diff
@@ -25,6 +25,17 @@ pub mod zoned_datetime;
 /// let date: DateTime<Gregorian> = parse_gregorian_from_str("2020-10-14T13:21:00")
 ///     .expect("Failed to parse a datetime.");
 /// ```
+///
+/// Optionally, fractional seconds can be specified: `YYYY-MM-DDThh:mm:ss.SSSS`.
+///
+/// ```
+/// use icu::datetime::mock::parse_gregorian_from_str;
+/// use icu_calendar::{DateTime, Gregorian};
+///
+/// let date: DateTime<Gregorian> = parse_gregorian_from_str("2020-10-14T13:21:00.101")
+///     .expect("Failed to parse a datetime.");
+/// assert_eq!(u32::from(date.time.fraction), 101);
```

Comment:
Suggestion: It would be nice to mention the unit (nanoseconds) in the assertion, since this is easy to be confused what it represents.

Reply:
nitpick (optional): Depending on how you address sffc's comment, `101_000_000` would be more readable than `100e6` for user-facing documentation for the less math-focused users.

Reply:
We've renamed fraction to nanosecond so this should be clear now, I agree it was ambiguous before. I'll use the more readable number representation :)